core/chain.py

```
@@ -1,22 +1,92 @@
+                ],
+                "state": self.state.to_dict() if hasattr(self.state, 'to_dict') else {}
+            }
+
+            temp_file = self._chain_file + ".tmp"
+            with open(temp_file, 'w') as f:
+                json.dump(data, f, indent=2)
+
+            os.replace(temp_file, self._chain_file)
+            logger.info(f"Saved chain with {len(self.chain)} blocks to {self._chain_file}")
+
+        except Exception as e:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)